geode-core/src/main/java/org/apache/geode/internal/cache/MemberFunctionStreamingMessage.java

```diff
@@ -188,7 +190,7 @@ protected void process(final ClusterDistributionManager dm) {
       FunctionContextImpl context = new MultiRegionFunctionContextImpl(cache,
           this.functionObject.getId(), this.args, resultSender, regions, isReExecute);
 
-      long start = stats.startTime();
+      start = stats.getTime();
       stats.startFunctionExecution(this.functionObject.hasResult());
```

Comment:
I don't like the inconsistent timings either.

Do we want to count the "result sender prep" work as part of function execution? If so, let's _always_ include the duration in the timer. That is, let's always get the start time and call `startFunctionExecution()` outside the `try`.

If we _don't_ want to count that prep work as part of function execution, it seems to me we also don't want to count prep failures as function execution failures.

I can think of two ways to avoid counting prep failures as function execution failures.

First:
- `boolean startedExecution = false;` before the `try` block.
- startedExecution = true` just before or after `stats.startFunctionExecution(…)`.
- Then in each catch block, record the failure only if `startedExecution` is `true`.

Second:
- Put the result sender prep work in one try/catch block, which never reports execution start and end.
- Put the actual function execution in a second try/catch block, which always reports execution start and end.

The wisdom of this second approach depends on how many catch blocks are needed for exceptions thrown by the result sender prep work.

Reply:
I removed the second call to `getTime()` so now successful executions will include the time to create the context, etc in the total time. This is something I will ask about on the dev/user list.

To avoid counting prep failures as execution failures, I went with @demery-pivotal's first suggestion above.